### ISSUE TITLE
"az iot edge deployment" stopped working recently, from Azure DevOps

### ISSUE THREAD

**Issue by FMauroy** [Azure/azure-cli#17555]
**Describe the bug**
We have been using the "az iot edge" CLI command for a while now to deploy our iot edge modules from an Azure DevOps release pipeline, but since yesterday, we receive strange errors. See below the log file, generated with the "--debug" option.
I tried to use an older version of "azure-iot", to no avail. At one point, I also saw messages asking me to use "az login", so I added the use of a service principal to get rid of these errors, but that didn't change anything regarding the other ones.
It looks like it has troubles with a Python module called paho-mqtt, so I added a pip install for it at the beginning, it installs successfully but the error still appears further down the processing.
I'm at a loss as to why this happens, would anyone have any idea? This has been working for months without any hiccup...
Thanks a lot!

**To Reproduce**
Here is the script ("Command line" template) I use to deploy the module:
`call az --version
call az login --service-principal --username %appId% --password %appSecret% --tenant %tenant%

rem easy_install -U setuptools
pip install --upgrade pip
pip install wheel
pip install paho-mqtt==1.5.0

echo Adding Edge extensions...
call az config set extension.use_dynamic_install=yes_without_prompt

rem call az extension add --name azure-iot --debug
call az extension add  --source https://github.com/Azure/azure-iot-cli-extension/releases/download/v0.10.8/azure_iot-0.10.8-py3-none-any.whl --debug

call az extension list

echo Deleting existing layered deployment config '%deploymentName%'...
call az iot edge deployment delete -d "%deploymentName%" --login "%iotHubConnectionString%" --debug

echo Deploying layered deployment config '%templateFilePath%'...
call az iot edge deployment create --content "%templateFilePath%" -d "%deploymentName%" --layered true --pri 5 -t "tags.environment = 'prod'" --login "%iotHubConnectionString%"

echo Done`

**Expected behavior**
No errors ;-)

**Environment summary**
Azure DevOps release pipeline, running on Windows 2016 with VS2017.

[tasklog_6.log](https://github.com/Azure/azure-cli/files/6250129/tasklog_6.log)

**Comment by yonzhan** [comment]
route to service team

**Comment by fma-wizata** [comment]
Hi all, as it was preventing a deployment in production, I downloaded the artefacts on my laptop and ran the main commands ("az iot edge deployment delete" and "az iot edge deployment create") and it worked without any issue. So this clears the CLI at first glance. Maybe the VM used in DevOps is wrongly configured? This is not something we can change, we can only select the VM in the list, and this is the only one with Windows and Visual Studio: "vs2017-win2016"
Out of curiosity, I tried "windows-2019" but it gives the same results.
For the time being I'll deploy from my laptop but that defeats the purpose of CI/CD ;-)

**Comment by miketimofeev** [comment]
Hi @fma-wizata!
I'm from the team owning hosted VMs administration and deployment. We're investigating the issue in our repo here as well https://github.com/actions/virtual-environments/issues/3143
Azure Cli has its own Python distribution, but the latest az-cli version seems to have a lack of setuptools and wheel python modules. The workaround is to install these modules in runtime via powershell:
```
$azPython = "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe"
& $azpython -m pip install setuptools
& $azpython -m pip install install wheel
```

**Comment by FMauroy** [comment]
Priviets @miketimofeev,
it worked, thanks a lot :)
Don't know why my attempts didn't, I already tried to install setuptools and wheel (but not install, maybe that's it).
Thanks and best regards
fred

**Comment by miketimofeev** [comment]
@FMauroy you've tried to install it using the default pip and not the one that comes along with azure-cli.
I think the issue should remain open because it's only a workaround, but the root cause needs to be fixed from the azure-cli side.

**Comment by jiasli** [comment]
The MSI installer of Azure CLI does include `setuptools` and `wheel`:

```
& "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe" -m pip list
setuptools             56.0.0
wheel                  0.36.2
```

This issue has been fixed by #17432 since 2.22.0 (https://github.com/actions/virtual-environments/issues/3143#issuecomment-822319693).

If you are still seeing this issue with the latest version (2.25.0), please kindly let us know.
